### PR TITLE
Require file size on file upload.

### PR DESCRIPTION
**Pull request by MrCreosote**
Required for blobstore compatibility.

Ran the TryShock code against blobstore.

### CODE REVIEW

File: src/us/kbase/shock/client/BasicShockClient.java [hunk 1]
```diff
@@ -406,14 +406,19 @@ public InputStream getFile(final ShockNode sn)
...
+	 * @param fileLength the length of the file in bytes.
 	 * @param filename the name of the file.
 	 * @param format the format of the file, e.g. ASCII, UTF-8, JSON. Ignored
 	 * if null or whitespace only.
 	 * @return a shock node object.
 	 * @throws IOException if an IO problem occurs.
 	 * @throws ShockHttpException if the node could not be created.
 	 */
-	public ShockNode addNode(final InputStream file, final String filename, final String format)
+	public ShockNode addNode(
```
Comment (sychan) [inline src/us/kbase/shock/client/BasicShockClient.java]: Just a style question - when you add new params, you put them in the middle of the existing params?

Comment (MrCreosote) [reply src/us/kbase/shock/client/BasicShockClient.java]: yep, doesn't really matter. Any code that uses this method won't compile either way, and the code will have to be update either way.

Comment (MrCreosote) [reply src/us/kbase/shock/client/BasicShockClient.java]: Also the file size relates most closely to the file data, so I wanted them next to each other.

File: src/us/kbase/shock/client/test/ShockTests.java [hunk 2]
```diff
@@ -761,7 +764,7 @@ public void invalidFileRequest() throws Exception {
 		}
 		sn.delete();
 		
-		sn = addNode(BSC1, "Been shopping? No, I've been shopping", "filename",
+		sn = addNode(BSC1, "Been shopping? No, I've been shopping", 27, "filename",
```
Comment (sychan) [inline src/us/kbase/shock/client/test/ShockTests.java]: What?

Comment (MrCreosote) [reply src/us/kbase/shock/client/test/ShockTests.java]: https://getyarn.io/yarn-clip/58192a42-0842-4801-8125-1969ecfab9b7

File: src/us/kbase/shock/client/BasicShockClient.java [hunk 3]
```diff
@@ -422,14 +427,17 @@ public ShockNode addNode(final InputStream file, final String filename, final St
 			throw new IllegalArgumentException(
 					"filename may not be null or empty");
 		}
+		if (fileLength < 0) {
+			throw new IllegalArgumentException("fileLength may not be negative");
```
Comment (sychan) [inline src/us/kbase/shock/client/BasicShockClient.java]: Is there a particular exception thrown for when the actual fileLength doesn't match the claimed fileLength? I didn't see one looking at the code, did I just miss it? The docs say the file will fail to save, but a descriptive error would be useful to developers.

Comment (MrCreosote) [reply src/us/kbase/shock/client/BasicShockClient.java]: Shock doesn't care, but the blobstore will throw an error that says the content length is incorrect. That'll wind up as a `ShockHTTPException` here.

Comment (MrCreosote) [reply src/us/kbase/shock/client/BasicShockClient.java]: I ran the TryShock code against the blobstore with a bad length and it was pretty clear what the problem was.

**Pull request merged**